src/cloud-providers/cmd/config-extractor/README.md

```diff
@@ -0,0 +1,79 @@
+# config-extractor
+
+A development tool for extracting and listing all configuration parameters for each cloud provider.
+
+## Purpose
+
+This tool parses the `manager.go` files for each cloud provider and extracts all flag definitions (command-line parameters and environment variables) to help developers and users understand the available configuration options.
+
+## Building
+
+From the `cloud-providers` directory:
+
+```bash
+make config-extractor
+```
+
+## Usage
+
+```bash
+./config-extractor [-o json|table] <provider-name>
+```
+
+### Arguments
+
+- `-o`: Output format (default: `json`)
+  - `json`: Outputs flags in JSON format
+  - `table`: Outputs flags in a formatted table
```

Comment:
What is the use-case for the table option? Do we need it as we are not testing it?

Reply:
The table option is for human readability... While AI is almost replacing me, I still prefer tables over JSON. :)

I'm not testing it because it's just pretty-printing the same data the JSON test already validates. If the parser works, the table works, testing behavior, not presentation.